site/docs/components/password_input.mdx

```diff
@@ -0,0 +1,97 @@
+---
+name: Password input
+menu: Components
+route: /components/password-input
+---
+
+import { Playground } from "docz";
+import { StatusLabel } from "hds-react";
+
+import ColorBox from "../../src/components/ColorBox";
+import LargeParagraph from "../../src/components/LargeParagraph";
+import Link from "../../src/components/Link";
+import Text from "../../src/components/Text";
+
+# Password input
+
+<StatusLabel type="alert">Draft</StatusLabel>
+<StatusLabel type="success" style={{marginLeft: 'var(--spacing-xs)'}}>Accessible</StatusLabel>
+
+<LargeParagraph>
+    A password field is a foundational input field component that the user can interact with and specifically input password data.
+</LargeParagraph>
+
+## Principles
+- A label should always be provided with password fields.
+- If possible, allow the password input to reflect the strength of the password entered.
+- If possible, add other programmatic assistance. Detect and suggest changes to provide the user with assistance in generating strong passwords.
+- If possible, do input validation in client-side real-time and provide the user with immediate feedback. See HDS Form validation pattern page for more information and best practices with password fields.
+- Password fields should always be mandatory and it is recommended to mark it as a required field.
+- Use `autocomplete="new-password"` and `id="new-password"` for the password input in a sign-up form, and for the new password in a reset-password form.
```

Comment:
@VilleMiekkoja @minevala Nämä taitavat olla sellaisia, jotka on jo otettu huomioon toteutuksessa itsessään? Tarvitseeko näitä erikseen mainita siis?

Reply:
IIRC, I think this is something HDS user has to configure. I did not see any sign of these in the implementation 🤔

Reply:
Then the next question is, should they be done under the hood if they are always required?

Reply:
Good question. I think there might be some use cases where autocomplete is not wanted: for example, a new password field or in a confirm action.

Reply:
Then the autocomplete attribute should be configurable: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete#values

Reply:
I've seen that some login forms do not use autocomplete because of security reasons. People using a shared machine could log in to someone else's account.

Reply:
Good points! I think it should be left for the HDS user to configure. As Mika stated there can be cases where different behaviour is wanted and therefore we should let users to decide.

Reply:
Our "gold standard" https://github.com/City-of-Helsinki/ASVS/blob/helsinki-4.0.1/4.0/en/0x11-V2-Authentication.md#user-content-v4.0.1-2.1.11 and specifically point 2.1.11 states we must 'Verify that "paste" functionality, browser password helpers, and external password managers are permitted.'. OTOH: MDN states that modern browsers disregard autocomplete=off for password management. https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#the_autocomplete_attribute_and_login_fields

I would just like to point out this high level requirement and I'm unsure  whether the autocomplete attribute actually enables or prevents this requirement from being satisfied.